src/components/Calculator.js

```diff
@@ -1,44 +1,73 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-// eslint-disable import/no-extraneous-dependencies
+import calculate from '../logic/calculate';
 
-// eslint-disable-next-line react/prefer-stateless-function
 class Calculator extends React.Component {
-  // eslint-disable-next-line no-useless-constructor
   constructor(props) {
     super(props);
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+    this.handleClick.bind(this);
   }
 
+  handleClick = (e) => {
+    const buttonName = e.target.textContent;
+    const { total, next, operation } = this.state;
+    const obj = calculate(({ total, next, operation }), buttonName);
+
+    this.setState(() => ({
+      total: obj.total,
+      next: obj.next,
+      operation: obj.operation,
+    }));
```

Comment:
- [ ] As the previous reviewer requested, kindly add useState to your app. You need to practice what you have learned about state/useState. The project [requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/project_events.md#:~:text=Note%3A%20Practice%20what%20you%20have%20learned%20until%20now%3A%20use%20useState%20and%20props.) that you submitted.